staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -481,17 +559,21 @@ func structToUnstructured(sv, dv reflect.Value) error {
 		fieldInfo := fieldInfoFromField(st, i)
 		fv := sv.Field(i)
 
+		if !fv.IsValid() {
```

Comment:
I missed this is the earlier review, but if the field name is `-` the field should be skipped.  I don't see that in the code, but we use it in a few places.